nio_cli/commands/blockcheck.py

```diff
@@ -0,0 +1,178 @@
+        self.check_version()
+
+        # What should I check for in regards to naming specifications?
+        self.print_check('class and file name')
+        self.check_naming()
+
+    def check_pep8(self):
+
+        shell_pep8 = 'pep8 .'
+        subprocess.call(shell_pep8, shell=True)
+        print('')
+        # Any way to grab pep8 output?
+
+    def check_spec(self):
+
+        if os.path.exists('spec.json'):
+            with open('spec.json') as f:
+                spec_dict = json.load(f)
+                self.blocks_in_spec = [
+                    k.split('/')[1] for k, v in spec_dict.items()
+                ]
+
+            if len(self.blocks_in_spec) > len(self.block_files):
+                print('There\'s extra blocks in the spec file')
+                sys.exit()
+            if len(self.blocks_in_spec) < len(self.block_files):
+                print('Not all blocks are in the spec file')
+                sys.exit()
+
+            for block in self.blocks_in_spec:
+                keys = ['version', 'description', 'properties', 'inputs',
+                        'outputs', 'commands']
+                for key in keys:
+                    if key not in [k for k in spec_dict['nio/' + block]]:
+                        print('{} block is missing {}'.format(block, key))
+                        sys.exit()
+                self.spec_versions_dict[block] = \
+                    spec_dict['nio/' + block]['version']
+
+            for key in ['version', 'description', 'properties']:
+                if not spec_dict['nio/' + block][key] \
+                        or spec_dict['nio/' + block][key] == '':
+                    print('Please fill in the {} of the {} block'.format(
+                        key, block))
+                    sys.exit()
+
+                for prop, val in \
+                        spec_dict['nio/' + block]['properties'].items():
+                    if val['description'] == '':
+                        print(
+                            'Please fill in the description for the '
+                            '"{}" property in the {} block'.format(prop, block)
+                        )
+                        sys.exit()
+        else:
+            print(
+                'Please run `nio buildspec {}` '
+                'from the project directory'.format(self._block)
+            )
+            sys.exit()
+        print('')
+
+    def check_readme(self):
+
+        if os.path.exists('README.md'):
+            with open('README.md') as f:
+                lines = [l.rstrip() for l in f.readlines()]
+            block_indices = []
+            for block in self.blocks_in_spec:
+                if block not in lines:
+                    print('Please add the {} block to the README')
+                block_indices.append(lines.index(block))
+            block_indices.sort()
+
+            for i in range(len(block_indices)):
+                # TODO: better way to handle `i+1` IndexError?
```

Comment:
Changing your range to explicitly include 0 and the last expected value +1 might fix this:

`range(0, len(block_indices)+1)`